packages/frontend-2/components/server-management/Table.vue

```diff
@@ -0,0 +1,106 @@
+        :style="{ paddingRight: paddingRightStyle }"
+      >
+        <div
+          v-for="header in headers"
+          :key="header.id"
+          :class="columnClasses[header.id]"
+          class="capitalize"
+        >
+          {{ header.title }}
+        </div>
+      </div>
+      <div
+        class="divide-y divide-outline-3 h-full overflow-visible"
+        :class="{ 'pb-32': overflowCells }"
+      >
+        <div
+          v-for="item in items"
+          :key="item.id"
+          class="relative grid grid-cols-12 items-center gap-6 px-4 py-1 min-w-[900px] bg-white"
+          :style="{ paddingRight: paddingRightStyle }"
+          :class="{ 'cursor-pointer hover:bg-primary-muted': !!props.onRowClick }"
+          tabindex="0"
+          @click="handleRowClick(item)"
+          @keydown.enter.space="handleRowClick(item)"
+        >
+          <template v-for="(column, colIndex) in headers" :key="column.id">
+            <div :class="getClasses(column.id, colIndex)" tabindex="0">
+              <slot :name="column.id" :item="item">
+                <div class="text-gray-900 font-medium order-1">
+                  {{ (item as any)[column.id] }}
+                </div>
+              </slot>
+            </div>
+          </template>
+          <div class="absolute right-0 flex items-center p-0 pr-0.5">
+            <div v-for="(button, btnIndex) in buttons" :key="btnIndex" class="p-1">
+              <FormButton
+                :icon-left="button.icon"
+                size="sm"
+                color="secondary"
+                hide-text
+                class="text-red-500"
+                @click.stop="button.action(item)"
+              />
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ConcreteComponent, computed } from 'vue'
+import {
+  ItemType,
+  UserItem,
+  ProjectItem,
+  InviteItem
+} from '~~/lib/server-management/helpers/types'
+
+type OnRowClickType = (item: ItemType) => void
+
+interface RowButton {
+  icon: ConcreteComponent
+  label: string
+  action: (item: ItemType) => void
+}
+
+interface Header {
+  id: string
+  title: string
+}
+
+const props = defineProps<{
+  headers: Header[]
+  items: Array<UserItem | ProjectItem | InviteItem>
+  buttons?: RowButton[]
+  columnClasses: Record<string, string>
+  overflowCells?: boolean
+  onRowClick?: OnRowClickType
+}>()
+
+const paddingRightStyle = computed(() => {
+  const padding = 52 + ((props.buttons as RowButton[]).length - 1) * 25
+  return `${padding}px`
+})
+
+const getClasses = (column: string, colIndex: number): string => {
+  const columnClass = props.columnClasses[column]
+
+  if (colIndex === 0) {
+    return `bg-transparent py-3 pr-5 ${columnClass}`
+  }
+  return `lg:p-0 ${columnClass}`
+}
+
+const handleRowClick = (item: UserItem | ProjectItem | InviteItem) => {
+  props.onRowClick?.(item as ItemType)
```

Comment:
Unnecessary type cast, item is already `ItemType`. Additionally you can shorten the type signature for this function's only argument to just be `ItemType`